src/components/Api.js

```diff
@@ -0,0 +1,123 @@
+export default class Api {
+  constructor(apiSettings) {
+    this._apiSettings = apiSettings;
+    this._headers = apiSettings.headers;
+    this._baseUrl = apiSettings.baseUrl;
+    this._userInfoUrl = `${this._baseUrl}/users/me`;
+    this._cardsUrl = `${this._baseUrl}/cards`;
+    this._avatarUrl = `${this._userInfoUrl}/avatar`;
+  }
+
+  _checkResponse(response) {
+    if (response.ok) {
+      return response.json();
+    }
+    return Promise.reject(response);
+  }
+
+  // async _request(url, methodType, options) {
+  //   if (options) {
+  //     return await fetch(url, {
+  //       method: methodType,
+  //       headers: this._headers,
+  //       body: JSON.stringify(options),
+  //     });
+  //   } else {
+  //     return await fetch(url, {
+  //       method: methodType,
+  //       headers: this._headers,
+  //     });
+  //   }
+  // }
+
+  async getInitialCards() {
+    const res = await fetch(this._cardsUrl, {
+      headers: this._headers,
+    });
+    const cardsArray = await this._checkResponse(res);
+    console.log(cardsArray);
+    return cardsArray;
```

Comment:
**Could be improved**

`await` is not needed if you can simplify the code with `then`

See how it looks without `await`

```js
  getInitialCards() {
    return fetch(this._cardsUrl, {
      headers: this._headers,
    }).then(this._checkResponse);
  }

```